src/NuGet.Core/NuGet.Frameworks/NuGetFrameworkFactory.cs

```diff
@@ -196,18 +205,30 @@ public static NuGetFramework ParseFolder(string folderName, IFrameworkNameProvid
                             || mappings.TryGetVersion(parts.Item2, out version))
                         {
                             var profileShort = parts.Item3;
-                            
+
                             if (version.Major >= 5
                                 && StringComparer.OrdinalIgnoreCase.Equals(FrameworkConstants.FrameworkIdentifiers.Net, framework))
                             {
                                 // net should be treated as netcoreapp in 5.0 and later
                                 framework = FrameworkConstants.FrameworkIdentifiers.NetCoreApp;
                                 if (!string.IsNullOrEmpty(profileShort))
                                 {
-                                    bool validProfile = FrameworkConstants.FrameworkProfiles.Contains(profileShort);
-                                    if (validProfile)
+                                    // Find a platform version if it exists and yank it out
+                                    var platformChars = profileShort;
```

Comment:
No need this extra platformChars variable here, keep using profileShort unless we do profileShort.ToCharArray(), but most likely no need;